src/AutofacSerilogIntegration/ContextualLoggingModule.cs

```diff
@@ -76,9 +80,9 @@ protected override void Load(ContainerBuilder builder)
                         .FirstOrDefault(np => np.Name == TargetTypeParameterName && np.Value is Type);
 
                     if (targetType != null)
-                        return (_logger ?? Log.Logger).ForContext((Type)targetType.Value);
+                        return (_loggerLambda(c) ?? Log.Logger).ForContext((Type)targetType.Value);
 
-                    return _logger ?? Log.Logger;
+                    return _loggerLambda(c) ?? Log.Logger;
```

Comment:
Here you're creating a logger. But who and where will dispose it?

Reply:
Normally, I would expect autofac to take care of this when disposing either root scope or nested lifetime scope. This kind of simplifies things. But you are using .ExternallyOwned() for the reasons I do not understand.

Reply:
(Not me, @nblumhardt )

The reasons are, I believe, quite simple. We're in this branch because `dispose` is `false`, which means the developer who passed the logger will be responsible for its disposal. Because it is only one instance, it is quite easy to achieve. But you change that to returning an unknown - possible new on every call! - logger instance from lambda, and that poses a question of disposal.

Please note that in your specific example the logger will not be (explicitly) disposed at all.